### PR TITLE
Visnu diet plan

### PR DESCRIPTION
**Pull request by visnusantos1203**
### [WIP]Diet Plan, Food Option, and API Wrapper

This PR is for Diet Plan feature where users will be able to create  custom diet plans for their specific fitness goals. We also created a food option database that will store data from users that's connected with the diet plan. The data from food option is going to be used for requesting recipes from the API so that users will be able to select from wide varieties of meals aligned with their fitness goals.

This PR includes: 

1. Form for creating Diet Plan
Diet Plan Form
2. Page where Diet Plans are displayed
Die Plan Page
3. Form for creating Food Option
Food Option Form
4. Page where data from Food Option are Displayed
Food Option Page 2
5. Lists of recipes from Edamam API that's been queried using the data from food option
Food Option Page
6. Rspec tests for Diet Plan model

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -30,6 +30,9 @@ Metrics/BlockLength:
     - "config/**/*"
     - "spec/**/*"
 
+Metrics/ParameterLists:
+  Enabled: false
```
Comment (analyn-cajocson) [inline .rubocop.yml:34]: why did you disable this? Revert this change.

File: app/api/edamam/request.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+module Edamam
+  class Request
+    APPLICATION_ID = 'cde8be74'.freeze
+    APPLICATION_KEY = '6746fb83b091d7a1339f2b70f85ef6fb'.freeze
```
Comment (analyn-cajocson) [inline app/api/edamam/request.rb:4]: Put this somewhere. eg. credentials.yml

File: app/views/diet_plans/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  <div>
+    <%= f.label :end_date %>
+    <%= f.date_field :end_date%>
+  </div>
+
+  <div>
+    <%= f.submit class: 'btn btn-outline-primary btn-sm' %>
+  </div>
+  <%= link_to 'Diet Plan', diet_plans_path, class: 'btn btn-primary btn-sm', method: :get %>
+<% end %>
```
Comment (analyn-cajocson) [inline app/views/diet_plans/_form.html.erb:21]: missing new line

File: app/views/diet_plans/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1 @@
+<%= render "form" %>
```
Comment (analyn-cajocson) [inline app/views/diet_plans/edit.html.erb:1]: missing new line

File: app/views/diet_plans/new.html.erb [hunk 5]
```diff
@@ -0,0 +1 @@
+<%= render "form" %>
```
Comment (analyn-cajocson) [inline app/views/diet_plans/new.html.erb:1]: missing new line

File: app/views/food_options/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    <p> Health Label: <%= option.health_label %> </p>
+    <p> Calories: <%= option.calories %> </p>
+    <p> Excluded: <%= option.excluded %> </p>
+
+  <% end %>
+</div>
+
+<%= link_to "Create new Food Option", new_food_option_path, method: :get %>
+
+<p> <%= @get_recipes %> </p>
```
Comment (analyn-cajocson) [inline app/views/food_options/index.html.erb:49]: missing new line

File: app/views/food_options/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  <div>
+    <%= f.label :excluded %>
+    <%= f.text_field :excluded%>
+  </div>
+
+  <div>
+    <%= f.submit class: 'btn btn-outline-primary btn-sm' %>
+  </div>
+  <%= link_to 'Diet Plan', diet_plans_path, class: 'btn btn-primary btn-sm', method: :get %>
+<% end %>
```
Comment (analyn-cajocson) [inline app/views/food_options/new.html.erb:51]: missing new line

File: .rubocop.yml (hunk 1 above)
Comment (analyn-cajocson) [reply .rubocop.yml:34]: Got it. Better to pass a hash parameter for this one instead of passing 9 parameters. That is a bad practice.

**Review by analyn-cajocson — CHANGES_REQUESTED** [review]

File: app/views/diet_plans/_form.html.erb (hunk 3 above)
Comment (visnusantos1203) [reply app/views/diet_plans/_form.html.erb:22]: resolved

**Review by analyn-cajocson — CHANGES_REQUESTED** [review]
Please rebase this against `master`
